downstairs/src/region.rs

```diff
@@ -424,6 +489,51 @@ fn config_path<P: AsRef<Path>>(dir: P) -> PathBuf {
     out
 }
 
+/**
+ * Remove directories associated with repair except for the replace
+ * directory. Replace is handled specifically during extent open.
+ */
+pub fn remove_copy_cleanup_dir<P: AsRef<Path>>(dir: P, eid: u32) -> Result<()> {
+    let mut remove_dirs = vec![copy_dir(&dir, eid)];
+    remove_dirs.push(completed_dir(&dir, eid));
+
+    for d in remove_dirs {
+        if Path::new(&d).exists() {
+            println!("Deleting dir: {:?}", d);
+            std::fs::remove_dir_all(&d)?;
+        }
+    }
+    Ok(())
+}
+
+/**
+ * Validate a list of repair files.
+ * There are either two or four files we expect to find, any more or less
+ * and we have a bad list.
+ */
+pub fn validate_repair_files(eid: usize, files: &[String]) -> bool {
```

Comment:
Do I understand this correctly that we expect the `files` list to be equal to either:

```rust
[
    extent_file_name(eid, None),
    extent_file_name(eid, Some("db"),
]
```
or 
```rust
[
    extent_file_name(eid, None),
    extent_file_name(eid, Some("db")),
    extent_file_name(eid, Some("db-shm")),
    extent_file_name(eid, Some("db-wal")),
]
```

If that's right would this make sense?

```rust
    let some = &[
        extent_file_name(eid, None),
        extent_file_name(eid, Some("db"),
    ].into_iter().collect::<BTreeSet<_>>();
    let mut all = some.clone();
    all.extend(&[
        extent_file_name(eid, Some("db-shm")),
        extent_file_name(eid, Some("db-wal")),
   ]);

    let these = files.iter().collect::<BTreeSet<_>>();

    // returns
    these == some || these == all
```
   
This may be overkill / not work at all for this use case (and my apologies), but I'd also note that this function seems fine with the case where e.g. `files` is 4x the same String.